### PR TITLE
added ShoppingCart model

### PR DESCRIPTION
**Pull request by furthernull**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @GetMapping
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        return shoppingCartService.getShoppingCart(authentication);
+    }
+
+    @Operation(summary = "Update cart item", description = "update existing item in shopping cart")
+    @PreAuthorize("hasAuthority('USER')")
+    @PutMapping("/items/{cartItemId}")
+    public CartItemDto update(@PathVariable Long cartItemId,
+            @RequestBody @Valid CartItemRequestDto requestDto,
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/controller/ShoppingCartController.java:53]:
```suggestion
            @RequestBody @Valid UpdateCartItemDto requestDto,
```

Create UpdateCartItemDto where int quantity

File: src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return cartItemMapper.toDto(cartItem);
+    }
+
+    @Override
+    public CartItemDto updateCartItem(Long cartItemId,
+                                      CartItemRequestDto requestDto,
+                                      Authentication authentication) {
+        Long userId = ((User) authentication.getPrincipal()).getId();
+        CartItem cartItem = cartItemRepository.findByIdAndShoppingCartId(cartItemId, userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't retrieve CartItem by id's"));
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java:47]:
```suggestion
                .map(item -> {
                    item.setQuantity(requestDto.quantity());
                    return item;
                }).orElseThrow(() -> new EntityNotFoundException(
                        String.format("No cart item with id: %d for user: %d", cartItemId, userId)
                ));
        cartItemRepository.save(cartItem);
        return cartMapper.toDto(cart);
```

File: src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Override
+    public CartItemDto addToCart(CartItemRequestDto requestDto, Authentication authentication) {
+        Long userId = ((User) authentication.getPrincipal()).getId();
+        ShoppingCart shoppingCart = shoppingCartRepository.getReferenceById(userId);
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(
+                () -> new EntityNotFoundException("Can't find Book by id: "
+                        + requestDto.bookId()));
+        cartItem.setShoppingCart(shoppingCart);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java:35]: What if shoppingCart already has the same book by Id, we need to increase quantity. 
If no, add cartItem to cart.

File: src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import com.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.example.bookstore.service.CartItemService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CartItemServiceImpl implements CartItemService {
+    private final BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/service/impl/CartItemServiceImpl.java:22]: Replace all logic to ShoppingCartServiceImpl

File: src/main/resources/db/changelog/changes/09-create-shopping-carts-table.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                    nullable: false
+                    foreignKeyName: fk_shopping_carts_users
+                    referencedTableName: users
+                    referencedColumnNames: id
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/09-create-shopping-carts-table.yaml:23]:
```suggestion
                    nullable: false
                    
```
Add empty line at the end of each changelog

File: src/main/java/com/example/bookstore/mapper/BookMapper.java [hunk 6]
```diff
@@ -59,4 +61,11 @@ default void setCategories(@MappingTarget Book book, BookRequestDto requestDto)
             = NullValuePropertyMappingStrategy.IGNORE)
     @Mapping(target = "categories", ignore = true)
     void updateModel(@MappingTarget Book book, BookRequestDto requestDto);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/bookstore/mapper/BookMapper.java:63]: Do you really need nullValuePropertyMappingStrategy for each field?

File: src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        shoppingCart.getCartItems().add(cartItem);
+        shoppingCartRepository.save(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto updateShoppingCart(Authentication authentication,
+                                              Long cartItemId,
+                                              CartItemUpdateQuantityDto requestDto) {
+        Long userId = ((User) authentication.getPrincipal()).getId();
```
Comment (okuzan) [inline src/main/java/com/example/bookstore/service/impl/ShoppingCartServiceImpl.java:77]: deal with it on controller level, don't pass authentication to the service, create a private method for repeated logic

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

**Review by Elena-Bruyako — APPROVED** [review]
Good job!

**Pull request merged**